### DISCUSSION TITLE
How to pass a pdf from UploadFile through some function?

### DISCUSSION THREAD

**Discussion by Nathan-Bernardo** [fastapi/fastapi#8790] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, APIRouter, status, Request, HTTPException, status, Body, File, UploadFile

@app.post("/resume/")
async def receive_file(file: UploadFile = File(...)):
    contents = await file.read()
    with open(f"/home/ncbernar/Github/listee/listeedemo/backend/app/resumes/{file.filename}", "wb") as f:
        f.write(contents)
    print(file.content_type)
    print(file.filename)
    print(file.file)
    return {"filename": file.filename}


@router.post("/resume/", response_description="Get resume from user")
async def create_candidate(file: UploadFile = File(...)):
    contents = await file.read()
    """
    want to pass the contents of the pdf to a parser defined in this function and pass the dictionary containing details of the document into another function.
    """
    return {"filename": file.filename}
```

### Description

* I am trying to pass a pdf file from the frontend to backend. 
* Initially, I had the file contents stored in my local system and have the document parser look through the local storge to find the resume.
* However, what I want is to pass the pdf file directly to the parser and have it parse the document.  
* The result should be a dictionary, which I want to pass to another function.
* I'm not exactly sure how I can do that.  Would anyone know what I can do? I even tried passing file.file into the function, but I end up getting this error: 
```
Reading from a temporary directory in python: TypeError: expected str, bytes or os.PathLike object, not TemporaryDirectory
```

### Operating System

Linux

### Operating System Details

Ubuntu 18.04

### FastAPI Version

0.61.1

### Python Version

3.8

**Comment by juansuv** [comment]
hi, try

from fastapi.responses import FileResponse

@app.get("/resume/", response_class=FileResponse)
async def create_candidate(filename:str):
    some_file_path=f'/home/ncbernar/Github/listee/listeedemo/backend/app/resumes/{filename}'
    return some_file_path

FileResponse does the magic

**Comment by YuriiMotov** [comment]
Questions is not clear
